landlab/components/species_evolution/species_evolver.py

```diff
@@ -0,0 +1,673 @@
+            returned in a list if this identifier exists. By default, taxa with
+            any identifier can be returned.
+        time : float, int, optional
+            Limit the taxa returned to those that existed at this time as
+            indicated in ``taxa_data_frame``. By default, taxa at all times in
+            the component record can be returned.
+        extant_at_latest_time : boolean, optional
+            Limit the taxa returned to taxa extant state at the latest time in
+            the record. By default, taxa with any extant state can be returned.
+        ancestor : int, optional
+            Limit the taxa returned to those descending from the taxon
+            designated as the ancestor. The ancestor is designated using its
+            ``uid``. By default, taxa with any or no ancestors are returned.
+
+        Returns
+        -------
+        taxa : a list of Taxon
+            The Taxon objects that pass through the filter. The list is sorted
+            by ``uid``. An empty list is returned if no taxa pass through the
+            filter.
+
+        Examples
+        --------
+        ZoneTaxon are used to demonstrate this method.
+
+        Import modules used in the following examples.
+
+        >>> from landlab import RasterModelGrid
+        >>> from landlab.components import SpeciesEvolver
+        >>> from landlab.components.species_evolution import ZoneController
+
+        Create a model grid with flat topography.
+
+        >>> mg = RasterModelGrid((3, 7), 1000)
+        >>> z = mg.add_ones('topographic__elevation', at='node')
+
+        Instantiate SpeciesEvolver and a ZoneController. Instantiate the
+        latter with a function that masks the low elevation zone extent. Only
+        one zone is created.
+
+        >>> se = SpeciesEvolver(mg)
+        >>> def zone_func(grid):
+        ...     return grid.at_node['topographic__elevation'] < 100
+        >>> zc = ZoneController(mg, zone_func)
+        >>> len(zc.zones) == 1
+        True
+
+        Introduce two taxa to the one zone.
+
+        >>> taxa = zc.populate_zones_uniformly(2)
+        >>> se.track_taxa(taxa)
+
+        Increment the model time, force mountain range formation to demonstrate
+        this method, and then advance model time again.
+
+        >>> z[[3, 10, 17]] = 200
+        >>> zc.run_one_step(1000)
+        >>> se.run_one_step(1000)
+
+        Remove the westernmost zone. Advance model time once again.
+
+        >>> z[mg.node_x < 3000] = 200
+        >>> zc.run_one_step(1000)
+        >>> se.run_one_step(1000)
+
+        Display taxa metadata.
+
+        >>> se.taxa_data_frame
+             appeared  latest_time  extant
+        uid
+        0           0         1000   False
+        1           0         1000   False
+        2        1000         2000   False
+        3        1000         2000    True
+        4        1000         2000   False
+        5        1000         2000    True
```

Comment:
Somehow it is very confusing to me that when you show all the data in the `taxa_data_frame` it does not show ancestor information, but then on line 545, one can get the ancestor information using `get_taxon_objects`. If I'm missing something that has a simple explanation, maybe add it.

Reply:
The data frame is to summarize taxa and get_taxon_objects with the ancestor parameter gets the objects that share that ancestor.

Also, taxa have multiple ancestors, so it wouldn't work well in this 2-d table. The ancestor in get taxon objects is a common ancestor. Parent uid could be added to table but its less crucial than other info. Decisions were made to minimize the amount of data added to the taxa data frame.

I'll consider explanations that don't add confusion.

Reply:
Got it. Don't worry about adding anything. It will probably just make things more confusing.